### PR TITLE
Rename Top to PartialSort, add OrderByDirection overloads

### PR DESCRIPTION
**Pull request by fsateler**
Rename Top/TopBy to PartialSort/PartialSortBy

Top somewhat implies expectations of sorting in descending order
(ie, the result would be the n largest elements).
PartialSort is a more neutral name, so use that.

Fixes: #187

PartialSort: Add overload taking an OrderByDirection parameter

### CODE REVIEW

**Comment by atifaziz** [comment]
I think you missed the rename of the [actual iterator method](https://github.com/morelinq/MoreLINQ/pull/188/commits/7ce9047d0948f7bb865d2292fec7f4bdcea4ece9#diff-0e4966bd1000cde6b315f043f2b16c7cR118).

File: MoreLinq/PartialSort.cs [hunk 1]
```diff
@@ -44,6 +44,27 @@ public static IEnumerable<T> PartialSort<T>(this IEnumerable<T> source, int coun
         }
 
         /// <summary>
+        /// Combines <see cref="OrderBy{T, TKey}(IEnumerable{T}, Func{T, TKey}, IComparer{TKey}, OrderByDirection)"/>,
```
Comment (atifaziz) [inline MoreLinq/PartialSort.cs]: The summary should mention a word about `direction` (without quoting the parameter). For example: “An additional parameter specifies the direction of the sort”. Ditto for other added overloads.

File: MoreLinq/PartialSort.cs [hunk 2]
```diff
@@ -67,6 +88,32 @@ public static IEnumerable<T> PartialSort<T>(this IEnumerable<T> source, int coun
...
+        /// where each element is its key, and <see cref="Enumerable.Take{TSource}"/>
+        /// in a single operation.
+        /// </summary>
+        /// <typeparam name="T">Type of elements in the sequence.</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="count">Number of (maximum) elements to return.</param>
+        /// <param name="comparer">A <see cref="IComparer{T}"/> to compare elements.</param>
+        /// <param name="direction">The direction on which to sort the elements</param>
```
Comment (atifaziz) [inline MoreLinq/PartialSort.cs]: > The direction on which to sort the elements

Should instead read, “The direction ~~on~~_in_ which to sort the elements.”

**Comment by atifaziz** [comment]
[README is missing the rename](https://github.com/morelinq/MoreLINQ/blame/a30936b18fa167b94313c61d469cbf0936567bc2/README.md#L77-L78).

**Comment by fsateler** [comment]
Good catch on the README. It is also missing in the nuspec file.

I have force pushed a new version with all your comments addressed.

**Comment by atifaziz** [comment]
> It is also missing in the nuspec file.

I see you found a catch too. 😉 Good eagle eyes. 😄 

Will review later and revert.

**Pull request merged**